projects/incubator/src/storefrontlib/shared/components/image-zoom/trigger/image-zoom-trigger.component.ts

```diff
@@ -0,0 +1,82 @@
+import {
+  Component,
+  ComponentRef,
+  Input,
+  OnDestroy,
+  OnInit,
+  ViewContainerRef,
+} from '@angular/core';
+import {
+  ICON_TYPE,
+  LaunchDialogService,
+  LAUNCH_CALLER,
+} from '@spartacus/storefront';
+import { combineLatest, Observable, Subscription } from 'rxjs';
+import { filter, tap } from 'rxjs/operators';
+import { ImageZoomDialogComponent } from '../dialog/image-zoom-dialog.component';
+
+@Component({
+  selector: 'cx-image-zoom-trigger',
+  styleUrls: ['image-zoom-trigger.component.scss'],
+  templateUrl: 'image-zoom-trigger.component.html',
+})
+export class ImageZoomTriggerComponent implements OnInit, OnDestroy {
+  iconType = ICON_TYPE;
+  protected subscriptions = new Subscription();
+
+  @Input() galleryIndex: number;
+  @Input() $expandImage: Observable<boolean>;
+
+  constructor(
+    protected launchDialogService: LaunchDialogService,
+    protected vcr: ViewContainerRef
+  ) {}
+
+  ngOnInit() {
+    // Used to open image by clicking the product in ProductImagesComponent
+    if (this.$expandImage) {
+      this.subscriptions.add(
+        this.$expandImage
+          .pipe(
+            filter((value) => Boolean(value)),
```

Comment:
As `value` is already a `boolean`, we can avoid using `Boolean()`

Reply:
Removed observable